packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -201,15 +208,32 @@ export const prepareAdvancerKit = (
             const { EUD } = decoded.query;
             log(`decoded EUD: ${EUD}`);
             assert.typeof(EUD, 'string');
-            // throws if the bech32 prefix is not found
-            const destination = chainHub.resolveAccountId(EUD);
+
+            const caipAddress = chainHub.makeCaipAddress(EUD);
+            const caip10Record = parseAccountId(caipAddress);
+            const chainId =
+              caip10Record.namespace === 'cosmos'
+                ? caipIdFromInfo(caip10Record)
+                : (`${caip10Record.namespace}:${caip10Record.reference}` as const);
```

Comment:
why remove the notice about the throwing behavior?

why give `destination` a new name when it's used multiple times below? you've removed the meaning of what the value is and replaced it with a description of its type.

```suggestion
            // throws if the bech32 prefix is not found
            const destination = chainHub.resolveAccountId(EUD);
            const destChain = chainOfAccount(destination);
```

Rebase onto master to get `chainOfAccount`.

Reply:
> why remove the notice about the throwing behavior?

That made sense when the previous code assumed it was a cosmos address. Now it can be CAIP-10 or Bech32, and throws if it's neither.  I've replaced it with a clearer warning.

> Rebase onto master to get chainOfAccount.

Thanks. I missed that getting added.